### ISSUE TITLE
Possibly a corner case for CLI tests?

### ISSUE THREAD

**Issue by omaciel** [SatelliteQE/robottelo#2899]
I noticed that the following test has failed today with latest downstream build:

* tests.foreman.cli.test_subnet.TestSubnet.test_negative_create_2

The test attempts to create a new subnet with incorrect values and I do see that there's an error displayed by the `hammer` command stating ``End of IP range must be specified if from is defined`` but it seems that the subnet is created as we can grab its info later on.

```bash
CLIFactoryError not raised
-------------------- >> begin captured logging << --------------------
robottelo: DEBUG: Running test TestSubnet/test_negative_create_2
robottelo.ssh: DEBUG: >>> [qe-sat6-rhel71.<EDITED>] LANG=en_US.UTF-8  hammer -v -u admin -p changeme --output=csv subnet create --from="195.74.101.88" --name="BPL9ZRs4" --network="195.74.101.212" --mask="255.255.255.0"
robottelo.ssh: INFO: Instantiated Paramiko client 0x7f5464079c90
robottelo.ssh: INFO: Destroying Paramiko client 0x7f5464079c90
robottelo.ssh: INFO: Destroyed Paramiko client 0x7f5464079c90
robottelo.ssh: DEBUG: <<< stderr
[ERROR 2015-10-13 15:37:18 Exception] End of IP range must be specified if from is defined
Could not create the subnet:
  End of IP range must be specified if from is defined

robottelo.ssh: DEBUG: >>> [qe-sat6-rhel71.<EDITED>] LANG=en_US.UTF-8  hammer -v -u admin -p changeme --output=csv subnet create --from="195.74.101.88" --name="T9wyPwSY" --network="195.74.101.212" --mask="255.255.255.0" --to="195.74.101.223"
robottelo.ssh: INFO: Instantiated Paramiko client 0x7f54640b3790
robottelo.ssh: INFO: Destroying Paramiko client 0x7f54640b3790
robottelo.ssh: INFO: Destroyed Paramiko client 0x7f54640b3790
robottelo.ssh: DEBUG: <<< stdout
Message,Id,Name
Subnet created,30,T9wyPwSY

robottelo.ssh: DEBUG: >>> [qe-sat6-rhel71.<EDITED>] LANG=en_US.UTF-8  hammer -v -u admin -p changeme  subnet info --id="30"
robottelo.ssh: INFO: Instantiated Paramiko client 0x7f5464079450
robottelo.ssh: INFO: Destroying Paramiko client 0x7f5464079450
robottelo.ssh: INFO: Destroyed Paramiko client 0x7f5464079450
robottelo.ssh: DEBUG: <<< stdout
Id:            30
Name:          T9wyPwSY
Network:       195.74.101.212
Mask:          255.255.255.0
Priority:      
DNS:           
Primary DNS:   
Secondary DNS: 
TFTP:          
DHCP:          
VLAN ID:       
Gateway:       
From:          195.74.101.88
To:            195.74.101.223
Domains:
```

I guess we may have 2 possible issues here:

1. The subnet should not have been created, which indicates an issue with `Foreman`
1. The `return code` for the creation command may be a zero (0) which would explain why the test also fails, as it expects a ** CLIFactoryError** error.

**Comment by omaciel** [comment]
@abalakh mind looking into this?

**Comment by abalakh** [comment]
It's not the issue with Foreman, something is wrong with test data generation.
The test verifies wrong `from` and `to` values, e.g. one of them unset or case when `from` is bigger than `to`. As you can see in log, `from` and `to` were provided, and both are valid. But we're expecting invalid value :)
I'll try to figure out what's going on and fix the test, thanks for pointing.

**Comment by abalakh** [comment]
@omaciel this should be resolved in #2903
